ddl/partition.go

```diff
@@ -154,3 +155,68 @@ func validRangePartitionType(col *table.Column) bool {
 		return false
 	}
 }
+
+// checkDropTablePartition checks if the partition exists and does not allow deleting the last existing partition in the table.
+func checkDropTablePartition(meta *model.TableInfo, partName string) error {
+	oldDefs := meta.Partition.Definitions
+	for _, def := range oldDefs {
+		if strings.EqualFold(def.Name.O, partName) {
+			if len(oldDefs) == 1 {
+				return errors.Trace(ErrDropLastPartition)
+			}
+			return nil
+		}
+	}
+	return errors.Trace(ErrDropPartitionNonExistent.GenByArgs(partName))
+}
+
+func removePartitionInfo(job *model.Job, t *meta.Meta, tblInfo *model.TableInfo, partName string) error {
+	oldDefs := tblInfo.Partition.Definitions
+	newDefs := make([]model.PartitionDefinition, 0, len(oldDefs)-1)
+	var pid int64
+	for i := 0; i < len(oldDefs); i++ {
+		if !strings.EqualFold(oldDefs[i].Name.O, partName) {
+			continue
+		}
+		pid = oldDefs[i].ID
+		newDefs = append(oldDefs[:i], oldDefs[i+1:]...)
+		break
+	}
+	job.TableID = pid
+	tblInfo.Partition.Definitions = newDefs
+	return nil
+}
+
+func onDropTablePartition(t *meta.Meta, job *model.Job) (ver int64, _ error) {
```

Comment:
I feel we need more comment for this function. At least, you need tell your read what are `ver` and `err`.